src/prefect/futures.py

```diff
@@ -404,10 +404,10 @@ def flow():
             print(f"Not Done: {len(not_done)}")
         ```
     """
-    futures = set(futures)
-    done = {f for f in futures if f._final_state}
-    not_done = futures - done
-    if len(done) == len(futures):
+    _futures = set(futures)
+    done = {f for f in _futures if f._final_state}
+    not_done = _futures - done
+    if len(done) == len(_futures):
```

Comment:
avoid overshadowing `futures` from signature, since mypy will expect the initially expressed type